app/lib/submission_builder/return_header.rb

```diff
@@ -3,6 +3,10 @@ class ReturnHeader < SubmissionBuilder::Document
     include SubmissionBuilder::FormattingMethods
     include SubmissionBuilder::BusinessLogicMethods
 
+    def state_submission_builder
```

Comment:
I think the original plan was to try and remove all `state_code` references entirely, including referencing the `StateInformationService`. However, that can't be done within the existing structure, since this `ReturnHeader` is shared between all states. It has no knowledge of what state it's representing without using state_code. The solution I chose uses the existing state_submission_builder attribute in the `StateInformationService` to figure out what state it's representing.

The other ideas you suggest both seem like viable alternatives!
1) Add to state intake, and then pull from intake here, rather than referencing the `StateInformationService`.

I prefer to avoid using intake, since conceptually, this doesn't match with how we utilize `intake`. `intake` is used for user input, while this is something static that appears on every NJ return, regardless of what the user selects.

2) Adding a separate NJ Return Header

This was the approach I originally had in mind. I'm not quite sure what it would take to implement though. I believe it would involve modifying the `NjReturnXml` child class to override the `return_header` method in the parent `StateReturn` class. We'd also need to create a new `NjReturnHeader` class that is essentially identical to `return_header.rb`, but adds the PaidPreparerFields. This approach feels most correct to me, but I recognize it would involve a lot of duplicated XML between the `ReturnHeader` and the `NjReturnHeader` classes.

Happy to discuss further with CfA about which option feels right here.

Reply:
@mmazanec22 I think you make a good point that this is a new pattern, and methods on the default and NJ intakes would be a better way to stay consistent but @jachanalso makes a good point that intakes are primarily for user input. Also, part of the refactor to implement `StateInformationService` was specifically to extract the static data configured in the intake classes to somewhere centralized. 

I think I was reluctant to add more fields to the `StateInformationService`, but after further consideration I think that is the correct home for state-specific static data such as this. Does that sound alright to you?